ebook.cpp

```diff
@@ -0,0 +1,79 @@
+#include <iomanip>
+#include <iostream>
+#include <vector>
+#include <utility>
+#include <fstream>
+#include <numeric>
+//Концепция решения: 
+/*
+
+Для решения задачи применим два вектора. Выбор именно векторов был осуществлен методом тыка с одной стороны и подглядывания в подсказку и решения других с другой,поэтому буду считать его оптимальным.
+
+Первый вектор это пользователи,где позиция в векторе это ID пользователя,а значение - количество страниц,прочитанных пользователем.
+
+Второй вектор это номера страниц где позиция в векторе это номер страницы,а значение это количество пользователей дочитавших до данной страницы.
+
+Мы изначально заполним оба вектора до максимальных значений нулями,так мы сможем обращаться к произвольному пользователю через индекс. 
+
+pages_read[0] - Так как нулевой страницы не бывает,используем эту переменную чтобы отображать сколько всего пользователей в данный момент читает книги.
+
+*/
+class Persons {
+public:
+	Persons(): persons_(std::vector<uint16_t>(100001, 0)), pages_read_(std::vector<uint32_t>(1001, 0)){ //Инициализируем векторы списком чтобы избежать промежуточного копирования
```

Comment:
Replace magic numbers in the code with named constants or enum values. For example, replace 100001 and 1001 with meaningful constant names.